qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,311 @@
+        commutators = [1j * commutator(operator, exc) for exc in self._excitation_pool]
+        res = estimate_observables(self.solver.estimator, self.solver.ansatz, commutators, theta)
+        return res
+
+    @staticmethod
+    def _check_cyclicity(indices: list[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> AdaptVQEResult:
+        """Computes the minimum eigenvalue.
+
+        Args:
+            operator: Operator whose minimum eigenvalue we want to find.
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than :class:`~VQE` or an ansatz other than
+                :class:`~.EvolvedOperatorAnsatz` is provided or if the algorithm finishes due
+                to an unforeseen reason.
+            AlgorithmError: If `quantum_instance` is not provided.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An :class:`~.AdaptVQEResult` which is a :class:`~.VQEResult` but also but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            termination criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
```

Comment:
Could we not get the ansatz from the solver this point and check it rather than doing it in the init. This will let the solver be updated/etc and here we get check/get the anstaz when we come to using VQE rather than doing it early.